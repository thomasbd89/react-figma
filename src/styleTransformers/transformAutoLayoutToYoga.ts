import { AutoLayoutProps, LayoutProps } from '../types';

const layoutModeToFlexDirection = {
    HORIZONTAL: 'row',
    VERTICAL: 'column'
};

const layoutAlignToAlignSelf = {
    MIN: 'flex-start',
    CENTER: 'center',
    MAX: 'flex-end',
    STRETCH: 'stretch'
};

const constraintToJustifyContent = {
    MIN: 'flex-start',
    CENTER: 'center',
    MAX: 'flex-end',
    STRETCH: 'space-between'
};

const constraintToAlignItems = {
    MIN: 'flex-start',
    CENTER: 'center',
    MAX: 'flex-end',
    STRETCH: 'stretch'
};

interface AutoLayoutAllProps extends AutoLayoutProps, LayoutProps {}

export const transformAutoLayoutToYoga = (props: AutoLayoutAllProps) => {
    return {
        ...(props.layoutMode && layoutModeToFlexDirection[props.layoutMode]
            ? {
                  flexDirection: layoutModeToFlexDirection[props.layoutMode]
              }
            : {}),
        ...(props.horizontalPadding
            ? {
                  paddingLeft: props.horizontalPadding,
                  paddingRight: props.horizontalPadding
              }
            : {}),
        ...(props.verticalPadding
            ? {
                  paddingTop: props.verticalPadding,
                  paddingBottom: props.verticalPadding
              }
            : {}),
        ...(props.layoutAlign && layoutAlignToAlignSelf[props.layoutAlign]
            ? {
                  alignSelf: layoutAlignToAlignSelf[props.layoutAlign]
              }
            : {}),
        ...(props.constraints &&
        props.constraints.horizontal &&
        constraintToJustifyContent[props.constraints.horizontal]
            ? {
                  justifyContent: constraintToJustifyContent[props.constraints.horizontal]
              }
            : {}),
        ...(props.constraints && props.constraints.vertical && constraintToAlignItems[props.constraints.vertical]
            ? {
                  alignItems: constraintToAlignItems[props.constraints.vertical]
              }
            : {}),
        ...(props.paddingLeft
            ? {
                  paddingLeft: props.paddingLeft
              }
            : {}),
        ...(props.paddingRight
            ? {
                  paddingRight: props.paddingRight
              }
            : {}),
        ...(props.paddingTop
            ? {
                  paddingTop: props.paddingTop
              }
            : {}),
        ...(props.paddingBottom
            ? {
                  paddingBottom: props.paddingBottom
              }
            : {})
    };
};
