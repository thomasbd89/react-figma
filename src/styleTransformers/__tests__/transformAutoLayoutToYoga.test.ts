import { transformAutoLayoutToYoga } from '../transformAutoLayoutToYoga';

describe('transformAutoLayoutToYoga', () => {
    it('constraints MIN/CENTER', () => {
        const result = transformAutoLayoutToYoga({
            constraints: {
                horizontal: 'MIN',
                vertical: 'CENTER'
            }
        });
        expect(result).toMatchSnapshot();
    });
});
