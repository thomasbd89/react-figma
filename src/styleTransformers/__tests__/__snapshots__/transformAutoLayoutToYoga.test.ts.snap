// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transformAutoLayoutToYoga constraints MIN/CENTER 1`] = `
Object {
  "alignItems": "center",
  "justifyContent": "flex-start",
}
`;
